.travis.yml

```diff
@@ -3,13 +3,20 @@ dist: trusty
 
 language: java
 jdk: oraclejdk8
+
+cache:
+  directories:
+  - $HOME/.m2
+  - $HOME/.p2
+  - extensions/ui/org.eclipse.smarthome.ui.paper/npm_cache
+
 before_install: echo "MAVEN_OPTS='-Xms1g -Xmx2g -XX:PermSize=512m -XX:MaxPermSize=1g'" > ~/.mavenrc
 
 #$ mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
 #$ mvn test -B
 
 install:
-  - echo 'mvn clean install -B -V -Dspotbugs.skip=true 1> .build.stdout 2> .build.stderr' > .build.sh
+  - echo 'mvn clean install -B -V -Dspotbugs.skip=true' > .build.sh
```

Comment:
When caching .m2, then it might be worth considering calling `verify` instead of `install` in oder to prevent the freshly built artifacts getting installed into the repository. Otherwise we'll end up having "ghosts" in there and risk building against old artifacts.

Reply:
Thanks for the hint, I was just about to remove the org.eclipse.smarthome artefacts from hand.